### DISCUSSION TITLE
Getting a list of allowed model for specific API key

### DISCUSSION THREAD

**Discussion by yaniv-fink** [betalgo/openai#268] [Q&A] [answered]
Hello,

I'm not sure if this feature has already been developed, but considering the fact that using an endpoint with a model without the appropriate permission results in an error, it would be helpful to provide developers/users with the ability to run tests before calling the endpoint. This way, they can verify if their API Key supports GPT-4 or not.

Another use case to consider is a developer managing multiple users, each with their own API keys and different permissions/models.

Thank you for your attention.

**Comment by Frogley** [comment, marked as answer]
You can use the [OpenAIService::ListModel](https://github.com/betalgo/openai/blob/master/OpenAI.SDK/Managers/OpenAIModel.cs#L9) function to query the models you have access to. Hope this helps.

**Reply by yaniv-fink** [reply]
Exactly what I was looking for, thank you!
